src/frontend/src/app/problem/problem.component.ts

```diff
@@ -81,6 +94,17 @@ export class ProblemComponent implements OnInit, OnDestroy {
 
   ngOnDestroy() {
     this.endDataSub?.unsubscribe();
+    this.currentTimeSub?.unsubscribe();
+    this.endTimeSub?.unsubscribe();
+  }
+
+  private setTimeDifference() {
+    const timeDifference = this.endTime.getTime() - new Date().getTime();
+    if (timeDifference <= 0) {
+      this.currentTimeSub?.unsubscribe();
+    }
+    this.timeLeftSeconds = Math.floor((timeDifference / 1000) % 60);
```

Comment:
Set it to a minimum of 0, just so that the UI doesn't go into the negatives

Reply:
Fixed now